cmd/juju/action/run.go

```diff
@@ -133,53 +133,54 @@ func (c *runCommand) SetFlags(f *gnuflag.FlagSet) {
 func (c *runCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "run-action",
-		Args:    "<unit> <action name> [key.key.key...=value]",
+		Args:    "<unit> [<unit> ...] <action name> [key.key.key...=value]",
 		Purpose: "Queue an action for execution.",
 		Doc:     runDoc,
 	}
 }
 
-// Init gets the unit tag, and checks for other correct args.
+// Init gets the unit tag(s), action name and action arguments.
 func (c *runCommand) Init(args []string) error {
-	switch len(args) {
-	case 0:
+	var unitNames []string
+	for idx, arg := range args {
+		if names.IsValidUnit(arg) {
```

Comment:
We can support any (or all) of the following:

Multiple units is easy, because if a token is a valid unit name, then it is an invalid action name and an invalid application name:
```sh
juju run-action unit/1 unit/2 action-name parameter list
juju run-action --units unit/1,unit/2 action-name parameter list
```
Running an action on all units (or just the leader) of an application is tricker, because a valid application name may also be a valid action name. We can support any (or all) of the following. Note that one format only allows a single application to be specified, which is probably fine but I can think of some more complex deploymements where I would want to run an action on 'all units in the cassandra-dc1 and cassandra-dc2 applications':

```
juju run-action application action-name parameter list
juju run-action application-a/* application-b/* action-name parameter list
juju run-action --all application-a --all application-b action-name parameter list
juju run-action --leader application-a --leader application-b action-name parameter list
```

My preferences would be:
```sh
juju run-action unit/1 unit/2 action-name parameter list
juju run-action --leader application-a action-name parameter list
juju run-action --all application-a --leader application-b --all application-c unit/1 unit/2 action-name parameter list
```

(although I'm unsure how easy it is to get the go command line parser to handle this)